### PR TITLE
[Juniper][QFX5210]Adding the system reboot handler

### PR DESCRIPTION
**Pull request by ciju-juniper**
The following changes are done as part of this commit:
 - Adding the system reboot handler
 - Adding swizzle reset case for the reboot reason
 - Workaround for the boot problem from Golden bios
 - Adding the logging messages for platform scripts
 - EEPROM parsing and library routines

### CODE REVIEW

**Comment by ciju-juniper** [comment]
@lguohan  Please merge these patches.

File: platform/broadcom/sonic-platform-modules-juniper/qfx5210/sonic_platform/chassis.py [hunk 1]
```diff
@@ -134,6 +258,8 @@ def get_reboot_cause(self):
                 return (ChassisBase.REBOOT_CAUSE_WATCHDOG, None)
             elif last_reboot_reason == "0x20":
                 return (ChassisBase.REBOOT_CAUSE_POWER_LOSS, None)
+            elif last_reboot_reason == "0x10":
+                return (ChassisBase.REBOOT_CAUSE_HARDWARE_OTHER, "Swizzle Reset")
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-juniper/qfx5210/sonic_platform/chassis.py]: what is swizzle reset?

File: platform/broadcom/sonic-platform-modules-juniper/debian/sonic-platform-juniper-qfx5210.postinst [hunk 2]
```diff
@@ -1,2 +1,16 @@
...
+
+if [ -f $FIRST_BOOT_FILE ]; then
+    mkdir /tmp/sda1
+    mount /dev/sda1 /tmp/sda1
+    cd /tmp/sda1/EFI
+    mkdir BOOT
+    cp SONiC-OS/grubx64.efi BOOT/BOOTX64.EFI
+    cd /tmp
+    umount sda1
+    efibootmgr -c -L "SONiC" -l "\EFI\BOOT\BOOTX64.EFI" > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (ciju-juniper) [reply platform/broadcom/sonic-platform-modules-juniper/debian/sonic-platform-juniper-qfx5210.postinst]: As I said in the description, there is a bios bug which fails to identify UEFI disk partition when the secondary bios is booted. Secondary bios is able to identify EFI/BOOT/BOOT64x.efi. This is a workaround till the bios bug is addressed.

[the remark this answers is not in the document]
Comment (lguohan) [reply platform/broadcom/sonic-platform-modules-juniper/debian/sonic-platform-juniper-qfx5210.postinst]: can you then add comments on this block of code?

[the remark this answers is not in the document]
Comment (ciju-juniper) [reply platform/broadcom/sonic-platform-modules-juniper/debian/sonic-platform-juniper-qfx5210.postinst]: @lguohan Done!

**Pull request merged**